src/pybind/mgr/dashboard/frontend/src/app/ceph/block/mirroring/pool-list/pool-list.component.ts

```diff
@@ -0,0 +1,173 @@
+import { Component, OnDestroy, OnInit, TemplateRef, ViewChild } from '@angular/core';
+
+import { I18n } from '@ngx-translate/i18n-polyfill';
+import { BsModalRef, BsModalService } from 'ngx-bootstrap/modal';
+import { Observable, Subscriber, Subscription } from 'rxjs';
+
+import { RbdMirroringService } from '../../../../shared/api/rbd-mirroring.service';
+import { CriticalConfirmationModalComponent } from '../../../../shared/components/critical-confirmation-modal/critical-confirmation-modal.component';
+import { CdTableAction } from '../../../../shared/models/cd-table-action';
+import { CdTableSelection } from '../../../../shared/models/cd-table-selection';
+import { FinishedTask } from '../../../../shared/models/finished-task';
+import { Permission } from '../../../../shared/models/permissions';
+import { AuthStorageService } from '../../../../shared/services/auth-storage.service';
+import { TaskWrapperService } from '../../../../shared/services/task-wrapper.service';
+import { PoolEditModeModalComponent } from '../pool-edit-mode-modal/pool-edit-mode-modal.component';
+import { PoolEditPeerModalComponent } from '../pool-edit-peer-modal/pool-edit-peer-modal.component';
+
+@Component({
+  selector: 'cd-mirroring-pools',
+  templateUrl: './pool-list.component.html',
+  styleUrls: ['./pool-list.component.scss']
+})
+export class PoolListComponent implements OnInit, OnDestroy {
```

Comment:
Since this is not my area of expertise (and I don't intend to become an expert), I was hoping that the modules would act like namespaces. I think it's going to be a terrible practice to ensure that each and every component has a unique name. You would essentially need to prepend the full directory hierarchy in each component name.

Reply:
They can have the same name, but than it's not so easy for the developer to know by the class name which class is mend ;)